### ISSUE TITLE
[MCHANGES-373] It is not possible to use "filter" parameter

### ISSUE THREAD

**Issue by jira-importer** [apache/maven-changes-plugin#243]
**[Angel Cervera Claudio](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=angelcervera)** opened **[MCHANGES-373](https://issues.apache.org/jira/browse/MCHANGES-373?redirect=false)** and commented

"filter" parameter is always ignored.

The reason is that this parameter is not set:
RestJiraDownloader.java line 145:
new JqlQueryBuilder( log ).urlEncode( false ).project( jiraProject ).fixVersion( getFixFor() ).fixVersionIds( resolvedFixVersionIds ).statusIds( resolvedStatusIds ).priorityIds( resolvedPriorityIds ).resolutionIds( resolvedResolutionIds ).components( resolvedComponentIds ).typeIds( resolvedTypeIds ).sortColumnNames( sortColumnNames ).build();

---

**Affects:** 2.12

**Issue Links:**
- [MCHANGES-308](https://issues.apache.org/jira/browse/MCHANGES-308) Support filters when using REST API / JQL
 (_**"is duplicated by"**_)

**Remote Links:**
- [GitHub Pull Request #91
](https://github.com/apache/maven-plugins/pull/91)

0 votes, 5 watchers

**Comment by jira-importer** [comment]
**[mikael petterson](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=eraonel)** commented

This issue can be fixed by adding the filter in:

String jqlQuery =
new JqlQueryBuilder( log ).urlEncode( false ).project( jiraProject ).fixVersion( getFixFor() ).fixVersionIds( resolvedFixVersionIds ).statusIds( resolvedStatusIds ).priorityIds( resolvedPriorityIds ).resolutionIds( resolvedResolutionIds ).components( resolvedComponentIds ).typeIds( resolvedTypeIds ).sortColumnNames( sortColumnNames ).filter(filter).build();

How can I submit a patch for it?

I forked form here:

https://github.com/apache/maven-plugins

And I have a made a local fix.

How do I proceed?

//mike
